lib/blocktree/blocktree_test.go

```diff
@@ -371,7 +371,82 @@ func TestBlockTree_IsDecendantOf(t *testing.T) {
 	require.False(t, isDescendant)
 }
 
-func TestBlockTree_HighestCommonAncestor(t *testing.T) {
+func TestNodeLowestCommonAncestor(t *testing.T) {
```

Comment:
this doesn't cover the error cases, please add test cases for these..

also do you think it could be re-arranged as test-cases + subtests?

Reply:
Actually with code changes I dont believe we can even hit the out of bounds error, so removing it

Reply:
actually take it back, will add error test case